packages/svelte/src/internal/client/reactivity/sources.js

```diff
@@ -197,6 +197,9 @@ function mark_reactions(signal, status) {
 		// Skip any effects that are already dirty
 		if ((flags & DIRTY) !== 0) continue;
 
+		// We can skip this reaction if it's MAYBE_DIRTY and either UNOWNED or if we're setting the status to MAYBE_DIRTY
+		if ((flags & (CLEAN | UNOWNED)) === 0 && status === MAYBE_DIRTY) continue;
```

Comment:
with this change (which I still find quite hard to parse if I'm honest!) the `if` on line 215 appears to be unnecessary:

https://github.com/sveltejs/svelte/blob/e352eccbc2fa19419a1b38797e78f3d7e692bf2b/packages/svelte/src/internal/client/reactivity/sources.js#L215

Reply:
Other conditions that behave equivalently, as far as the tests are concerned:

```suggestion
		if ((flags & CLEAN) === 0 && status === MAYBE_DIRTY) continue;
```

```suggestion
		if ((flags & status) !== 0) continue;
```

It's just really hard to tease apart the various elements here and understand when we mark and when we don't, and why.